src/cli/adapter/prettier.ts

```diff
@@ -0,0 +1,146 @@
+import fs from 'fs';
+import path from 'path';
+import { inspect } from 'util';
+
+import { check, format, getFileInfo, resolveConfig } from 'prettier';
+
+import { crawlDirectory } from '../../utils/dir';
+import { Logger } from '../../utils/logging';
+import { getConsumerManifest } from '../../utils/manifest';
+
+interface Result {
+  count: number;
+  errored: Array<{ err?: unknown; filepath: string }>;
+  touched: string[];
+  unparsed: string[];
+  untouched: string[];
+}
+
+const formatFile = async (
+  filepath: string,
+  logger: Logger,
+  mode: 'format' | 'lint',
+  result: Result,
+) => {
+  logger.debug(filepath);
+
+  const [config, data, fileInfo] = await Promise.all([
```

Comment:
I’m guessing this is using lower level helpers instead of `prettier.format` to give better user feedback? Is it worth leaving a comment to that effect?

Reply:
`prettier.format` is fairly low level itself actually and is synchronous. I've tried to clarify this a little in the file.